### PR TITLE
Use given format in DateTimeValidator instead of DATETIME_FORMAT

### PR DESCRIPTION
**Pull request by alireza-m**

### CODE REVIEW

File: src/Element/Month.php [hunk 1]
```diff
@@ -28,6 +28,13 @@ class Month extends DateTime
     ];
 
     /**
+     * A valid format string accepted by date()
+     *
+     * @var string
+     */
+    protected $format = self::DATETIME_FORMAT;
```
Comment (froschdesign) [inline src/Element/Month.php]: This is already defined in `Zend\Form\Element\DateTime`.

File: test/Element/DateTest.php [hunk 2]
```diff
@@ -128,14 +128,14 @@ public function testCorrectFormatPassedToDateValidator()
             'min'       => '2012-01-01',
             'max'       => '2012-12-31',
         ]);
-        $element->setFormat('d-m-Y');
+        $element->setFormat('Y-m-d');
```
Comment (froschdesign) [inline test/Element/DateTest.php]: Please undo this change. This test checks only the format and previous version was correct.

File: test/Element/DateTest.php [hunk 3]
```diff
@@ -128,14 +128,14 @@ public function testCorrectFormatPassedToDateValidator()
...
 
         $inputSpec = $element->getInputSpecification();
         foreach ($inputSpec['validators'] as $validator) {
             switch (get_class($validator)) {
                 case 'Zend\Validator\DateStep':
                 case 'Zend\Validator\Date':
-                    $this->assertEquals('d-m-Y', $validator->getFormat());
+                    $this->assertEquals('Y-m-d', $validator->getFormat());
```
Comment (froschdesign) [inline test/Element/DateTest.php]: Please undo this change.

File: src/Element/Month.php (hunk 1 above)
Comment (alireza-m) [inline src/Element/Month.php]: Yes it's defined there but in the parent method valueIsValidDateTimeFormat we should use $format instead of the constant

File: test/Element/DateTest.php (hunk 2 above)
Comment (alireza-m) [inline test/Element/DateTest.php]: If we want to change the format we have to change min and max values 
min value should be 01-02-2012
max value should be 12-31-2012

File: src/Element/Month.php (hunk 1 above)
Comment (froschdesign) [inline src/Element/Month.php]: Right. This means: change the parent not child.

File: test/Element/DateTest.php (hunk 2 above)
Comment (froschdesign) [inline test/Element/DateTest.php]: For this test the values are irrelevant.

**Comment by froschdesign** [comment]
Closing in favor of #172
